docs/docs/shards/types.md

```diff
@@ -0,0 +1,642 @@
+
+An **Int3** type value looks like this: `(int3 2 4 -4)`.
+
+[`(Math.Add)`](https://docs.fragcolor.xyz/shards/Math/Add/) is an example of a shard that uses this data type for its input, output, and `:Operand` parameter.
+
+=== "Code"
+
+    ```clojure linenums="1"
+    (int3 1 10 99) (Math.Add (int3 99 90 1))
+    (Log)   ;; int3 output => (100, 100, 100)
+    ```
+
+=== "Output"
+
+    ```
+    [info] [2022-07-22 21:24:38.132] [T-25580] [logging.cpp::55] [mywire] (100, 100, 100)
+    ```
+
+## Int4
+
+Type **Int4** is like type [**Int3**](#int3) but is a vector of four 32-bit signed integers instead.
+
+!!! note
+    Has keyword [`int4`](https://docs.fragcolor.xyz/functions/values/#int4) and alias `Int4`.
+
+An **Int4** type value looks like this: `(int4 1 -4 0 3)`.
+
+[`(Math.Add)`](https://docs.fragcolor.xyz/shards/Math/Add/) is an example of a shard that uses this data type for its input, output, and `:Operand` parameter.
+
+=== "Code"
+
+    ```clojure linenums="1"
+    (int4 3 6 9 4)
+    (Math.Add (int4 6 3 0 5))
+    (Log)   ;; int4 output => (9, 9, 9, 9)
+    ```
+
+=== "Output"
+
+    ```
+    [info] [2022-07-22 21:27:58.115] [T-20884] [logging.cpp::55] [mywire] (9, 9, 9, 9)
+    ```
+
+## Int8
+
+Type **Int8** defines a vector of eight 16-bit signed integers.
+
+16 bits of memory for each number allows this data type to store integer values ranging from −32,768 to +32,767 (no decimals).
+
+The shard `(Math.Add)` accepts **Int8** as input and as its `:Operand`. The shard adds these up outputs the sum as another vector of 8 integers or **Int8** data type. 
+
+## Int16
+
+Type **Int16** defines a vector of sixteen 8-bit signed integers.
+
+8 bits of memory for each number allows this data type to store integer values ranging from −128 to +127 (no decimals).
+
+The shard `(Math.Add)` accepts **Int16** as input and as its `:Operand`. The shard adds these up outputs the sum as another vector of 16 integers or **Int16** data type. 
+
+## None
+
+Type **None** indicates that no data type is expected. This implies that no value is expected.
+
+For example, **None** as one of the valid data types for `:Max` parameter in shard [`(RandomInt)`](https://docs.fragcolor.xyz/shards/General/RandomInt/) means that setting a value for this parameter is not mandatory.
+
+```clojure linenums="1"
+(RandomInt
+:Max [(None) (Int) (ContextVar [(Int)])]
+)
+```
+
+`(RandomInt)` generates a random integer and the `:Max` parameter is the upper limit (not inclusive) of the value that can be generated. So it makes sense to have **None** as one of the valid types for this `:Max` parameter for cases when you do not want an upper limit on the random integer.
+
+=== "Code"
+
+    ```clojure linenums="1"
+    (RandomInt 8)
+    (Log)   ;; max int that can be generated is 7
+    
+    (RandomInt)
+    (Log)   ;; no upper limit on the generated int
```

Comment:
Technically the maximum 64-bit value for a signed `int` is the limit. It is not exactly infinite.

Reply:
Fixed with commit [#161202abdf6bb7c5ecc8f2fb45d338408590b74b](https://github.com/fragcolor-xyz/shards/pull/351/commits/161202abdf6bb7c5ecc8f2fb45d338408590b74b)